From 3e356cce00fdae6da780613012c4c7be9047f19c Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 16 Jan 2017 14:00:29 +0000
Subject: [PATCH 0739/1000] MANTIS 44263 massaiCfgXmgrB.exe version 1.6.3 -
 MANTIS 44263: Platform can't be started after Kiosk reboot - Do not allow
 "illegal" characters in an item's data (explanation and value).

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1137 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h    | 14 +++++++++++--
 .../Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h | 14 +++++++++++--
 .../cpp/CfgXMgrB/src/cfgCorba_Selection.cpp   | 12 ++++++++++-
 .../Massai/cpp/CfgXMgrB/src/cfgToXML.cpp      | 21 +++++++++++++------
 4 files changed, 50 insertions(+), 11 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
index 014201cc..af045d6b 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/CfgXMgrB_def.h
@@ -474,8 +474,18 @@
 //! @par     Change:
 //! - MANTIS 43598: MassaiCfgXB: unexpected termination
 //! - Mutex-protect itemGet/itemSet sequences.
-#define FILE_VER_BIN 1,6,2,0
-#define FILE_VER_DTS "1.6.2.0"
+// #define FILE_VER_BIN 1,6,2,0
+// #define FILE_VER_DTS "1.6.2.0"
+//
+//! @version 1.6.3
+//! @date    16.01.2017
+//! @author  JK
+//! @par     Change:
+//! - MANTIS 44263: Platform can't be started after Kiosk reboot
+//! - Do not allow "illegal" characters in an item's data (explanation
+//!   and value).
+#define FILE_VER_BIN 1,6,3,0
+#define FILE_VER_DTS "1.6.3.0"
 //
 
 
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
index c2b5287c..20239af6 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/inc/massaiCfgXB_def.h
@@ -465,8 +465,18 @@
 //! @par     Change:
 //! - MANTIS 43598: MassaiCfgXB: unexpected termination
 //! - Mutex-protect itemGet/itemSet sequences.
-#define FILE_VER_BIN 1,6,2,0
-#define FILE_VER_DTS "1.6.2.0"
+// #define FILE_VER_BIN 1,6,2,0
+// #define FILE_VER_DTS "1.6.2.0"
+//
+//! @version 1.6.3
+//! @date    16.01.2017
+//! @author  JK
+//! @par     Change:
+//! - MANTIS 44263: Platform can't be started after Kiosk reboot
+//! - Do not allow "illegal" characters in an item's data (explanation
+//!   and value).
+#define FILE_VER_BIN 1,6,3,0
+#define FILE_VER_DTS "1.6.3.0"
 //
 
 //-------------------------------------------------------------------------
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgCorba_Selection.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgCorba_Selection.cpp
index faf1e5d5..502eec8b 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgCorba_Selection.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgCorba_Selection.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2002-2016 MATERNA Information & Communications
+ * Copyright (c) 2002-2017 MATERNA Information & Communications
  *
  **************************************************************************
 **/
@@ -822,6 +822,11 @@ private:
 cfgCorba_Selection::cfgCorba_Selection( cfgEntity & Configuration ) :
    Handle( new cfgCorba_Selection_Private( Configuration ) )
 {
+   writeTrace
+      (
+         SecondaryRuntime_HERE,
+         "cfgCorba_Selection::cfgCorba_Selection"
+      );
 }
 
 
@@ -830,6 +835,11 @@ cfgCorba_Selection::~cfgCorba_Selection()
 {
    try
    {
+      writeTrace
+         (
+            SecondaryRuntime_HERE,
+            "cfgCorba_Selection::~cfgCorba_Selection"
+         );
       delete Handle;
    }
    catch( ... )
diff --git a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgToXML.cpp b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgToXML.cpp
index 350a92d9..7cd72e16 100644
--- a/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgToXML.cpp
+++ b/SelfServiceCommon/Massai/cpp/CfgXMgrB/src/cfgToXML.cpp
@@ -7,7 +7,7 @@
  *
  * @author Juergen Kreierhoff
  *
- * Copyright (c) 2002-2007 Materna Information & Communications
+ * Copyright (c) 2002-2017 Materna Information & Communications
  *
  **************************************************************************
 **/
@@ -27,11 +27,20 @@ cfgToXML::cfgToXML( std::string const & Src ) :
 void cfgToXML::copy( std::ostream & Target ) const
 {
    char const * Src = M_Src->c_str();
-   char X = *Src++;
+   unsigned char X = *Src++;
    while( X )
    {
       switch( X )
       {
+      case 0x09:
+         Target << "&#x9;";
+         break;
+      case 0x0A:
+         Target << "&#xA;";
+         break;
+      case 0x0D:
+         Target << "&#xD;";
+         break;
       case '\'':
          Target << "&apos;";
          break;
@@ -48,18 +57,18 @@ void cfgToXML::copy( std::ostream & Target ) const
          Target << "&gt;";
          break;
       default:
-         if( ( X < ' ' ) || ( X > 'z' ) )
+         if( ( X > 'z' ) )
          {
-            unsigned char const Val = static_cast<unsigned char>( X );
             Target
                << "&#x"
-               << std::hex << static_cast<int>( Val ) << std::dec
+               << std::hex << static_cast<int>( X ) << std::dec
                << ";";
          }
-         else
+         else if( X >= ' ' )
          {
             Target << X;
          }
+         // No else - other characters below SPACE are ignored.
       }
       X = *Src++;
    }
-- 
2.41.0.windows.1

